agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/logging/DiagnosticTelemetryPipelineListener.java

```diff
@@ -56,47 +58,49 @@ public DiagnosticTelemetryPipelineListener(
   @Override
   public void onResponse(TelemetryPipelineRequest request, TelemetryPipelineResponse response) {
     int responseCode = response.getStatusCode();
-    switch (responseCode) {
-      case 200: // SUCCESS
-        operationLogger.recordSuccess();
-        break;
-      case 206: // PARTIAL CONTENT, Breeze-specific: PARTIAL SUCCESS
-      case 400: // breeze returns if json content is bad (e.g. missing required field)
-        operationLogger.recordFailure(
-            getErrorMessageFromPartialSuccessResponse(response.getBody(), responseCode));
-        break;
-      case 307:
-      case 308:
-        operationLogger.recordFailure("Too many redirects");
-        break;
-      case 401: // breeze returns if aad enabled and no authentication token provided
-      case 403: // breeze returns if aad enabled or disabled (both cases) and
-        if (!suppressWarningsOnRetryableFailures) {
+    try (MDC.MDCCloseable ignored = AzureMonitorMsgId.NETWORK_FAILURE_ERROR.makeActive()) {
```

Comment:
let's see if we can come up with cleaner solution than wrapping everything including success

maybe an overload on OperationLogger that takes a msgid?